lib/contentful_delivery/locales.ex

```diff
@@ -0,0 +1,82 @@
+defmodule Contentful.Delivery.Locales do
+  @moduledoc """
+  Handles the fetching of locales within a given `Contentful.Space`
+  """
+  alias Contentful.{Delivery, Locale, Space}
+
+  @doc """
+  Will attempt to fetch all locales for a given space
+
+  Does currently *not* accept collection parameters, as the API does not support them
+
+  ## Examples
+
+      space = "my_space_id"
+      space |> Locales.fetch_all()
+      {:ok,
+       [
+         %Contentful.Locale{
+           code: "en-US",
+           default: true,
+           fallback_code: nil,
+           name: "English (United States)"
+         },
+         %Contentful.Locale{
+           code: "de",
+           default: false,
+           fallback_code: "en-US",
+           name: "German"
+         }
+       ]}
+
+  """
+  @spec fetch_all(
+          String.t(),
+          Space.t() | String.t(),
+          String.t() | nil
+        ) ::
+          list(Locale.t())
+  def fetch_all(
+        space \\ Delivery.config(:space),
+        env \\ Delivery.config(:environment),
+        api_key \\ Delivery.config(:access_token)
+      )
+
+  def fetch_all(%Space{meta_data: %{id: space_id}}, env, api_key) do
+    space_id
+    |> build_request(env, api_key)
+    |> Delivery.send_request()
+    |> Delivery.parse_response(&build_locales/1)
+  end
+
+  def fetch_all(space_id, env, api_key) when is_binary(space_id) do
+    fetch_all(%Space{meta_data: %{id: space_id}}, env, api_key)
+  end
+
+  defp build_request(space, env, api_key) do
+    url = [
+      space |> Delivery.url(env),
+      "/locales"
+    ]
+
+    {url, api_key |> Delivery.request_headers()}
+  end
+
+  defp build_locales(%{"items" => items}) do
```

Comment:
If this gets public and more generalized (not passing a map, but the `items` list directly, it can be reused in `Spaces` below. In fact we care about the mapper in it. Maybe something like this could work?

```elixir
defp build_locales(%{"items" => items}) do
  {:ok, items |> build_locales_list()}
end

# also use this in Spaces. build_space/1
def build_locales_list(items) do
  items
  |> Enum.map(fn %{
                  "name" => name,
                  "code" => code,
                  "fallbackCode" => fallback_code,
                  "default" => default
                } ->
    %Locale{
      name: name,
      code: code,
      fallback_code: fallback_code,
      default: default
    }
  end)
end
```